### PR TITLE
[libc][math] Add option to set a specific exponent for frexp with Inf/NaN inputs.

### PR DESCRIPTION
**Pull request by lntue**
In IEEE 754 and C standards, when calling `frexp` with Inf/Nan inputs, the exponent result is unspecified.  In this case, FreeBSD libc and musl just passthrough `exp`, while glibc, FreeBSD libm set exp = 0, and MSVC set exp = -1.

By default, LLVM libc will passthrough `exp` just as FreeBSD libc and musl, but we also allow users to explicitly choose the return exp value in this case for compatibility with other libc.

Notice that, gcc did generate passthrough `exp` for `frexp(NaN/Inf, exp)`: https://godbolt.org/z/sM8fEej4E

### CODE REVIEW

**Comment by lntue** [comment]
Maybe I should keep the cmake config flag and the macro definition of the same name.

**Comment by nickdesaulniers** [comment]
Why do we want to make this configurable? Mind adding more context to the PR description?

**Comment by lntue** [comment]
> Why do we want to make this configurable? Mind adding more context to the PR description?

I updated the PR description to describe the issue.

File: libc/config/config.json [hunk 1]
```diff
@@ -87,6 +87,10 @@
     "LIBC_CONF_MATH_OPTIMIZATIONS": {
       "value": 0,
       "doc": "Configures optimizations for math functions. Values accepted are LIBC_MATH_SKIP_ACCURATE_PASS, LIBC_MATH_SMALL_TABLES, LIBC_MATH_NO_ERRNO, LIBC_MATH_NO_EXCEPT, and LIBC_MATH_FAST."
+    },
+    "LIBC_CONF_FREXP_INF_NAN_EXPONENT": {
+      "value": "",
+      "doc": "Set the specific exp value for Inf/NaN inputs."
```
Comment (nickdesaulniers) [inline libc/config/config.json:93]:
```suggestion
      "doc": "The value written back to the second parameter when calling frexp/frexpf/frexpl` with `+/-Inf`/`NaN` is unspecified.  Configue an explicit exp value for Inf/NaN inputs."
```

**Review by nickdesaulniers — APPROVED** [review]

File: libc/config/config.json (hunk 1 above)
Comment (lntue) [reply libc/config/config.json:93]: Done.

File: libc/src/__support/FPUtil/ManipulationFunctions.h [hunk 2]
```diff
@@ -31,8 +31,12 @@ namespace fputil {
 template <typename T, cpp::enable_if_t<cpp::is_floating_point_v<T>, int> = 0>
 LIBC_INLINE T frexp(T x, int &exp) {
   FPBits<T> bits(x);
-  if (bits.is_inf_or_nan())
+  if (bits.is_inf_or_nan()) {
+#ifdef LIBC_FREXP_INF_NAN_EXPONENT
```
[the remark this answers is not in the document]
Comment (lntue) [reply libc/src/__support/FPUtil/ManipulationFunctions.h:35]: Done.

**Pull request merged**